openmdao/core/tests/test_setup_memleak.py

```diff
@@ -41,9 +43,11 @@ def test_setup_memleak(self):
                 # Force garbage collection now instead of waiting for an
                 # optimal/arbitrary point since we're tracking memory usage
                 gc.collect()
-                snapshots.append(tracemalloc.take_snapshot())
+                if i == 0:
```

Comment:
Shouldn't we be taking the first snapshot *before* prob.setup?

Reply:
The way I read the existing code was that it checks that the memory footprint does not change with multiple setup calls, rather than between pre-setup and setup (times n)